web-extension-1/src/background/background.ts

```diff
@@ -1,5 +1,83 @@
-const setupBackground = async () => {
-  console.log('setup background');
-};
+import { Itab, currentTabPromise, IPopupEvent, IPortDisconnected } from '../interfaces/tab'
+
+let tabs:Itab[] = new Array<Itab>();
```

Comment:
a `Map` would have simplified the logic of the `findOrInitializeTab` function below